hudi-common/src/main/java/org/apache/hudi/common/table/read/buffer/DefaultFileGroupRecordBufferLoader.java

```diff
@@ -59,7 +60,23 @@ public Pair<HoodieFileGroupRecordBuffer<T>, List<String>> getRecordBuffer(Hoodie
                                                                             ReaderParameters readerParameters,
                                                                             HoodieReadStats readStats,
                                                                             Option<BaseFileUpdateCallback<T>> fileGroupUpdateCallback) {
+    Option<FileGroupRecordBuffer<T>> recordBufferOpt = getFileGroupRecordBuffer(readerContext, inputSplit, orderingFieldNames, hoodieTableMetaClient, props,
+        readerParameters, readStats, fileGroupUpdateCallback);
+    return recordBufferOpt.map(recordBuffer -> Pair.of(recordBufferOpt.get(), scanLogFiles(readerContext, storage, inputSplit, hoodieTableMetaClient, props,
+        readerParameters, readStats, recordBufferOpt.get())));
+  }
 
+  private Option<FileGroupRecordBuffer<T>> getFileGroupRecordBuffer(HoodieReaderContext<T> readerContext,
+                                                           InputSplit inputSplit,
+                                                           List<String> orderingFieldNames,
+                                                           HoodieTableMetaClient hoodieTableMetaClient,
+                                                           TypedProperties props,
+                                                           ReaderParameters readerParameters,
+                                                           HoodieReadStats readStats,
+                                                           Option<BaseFileUpdateCallback<T>> fileGroupUpdateCallback) {
+    if (inputSplit.getLogFiles().isEmpty()) {
+      return Option.empty();
+    }
```

Comment:
nit: no need to separate the method out since `DefaultFileGroupRecordBufferLoader` is not extended and there is no other usage of the method?